### ISSUE TITLE
Page endpoints not working when navigating with trailingSlash="always"

### ISSUE THREAD

**Issue by pzerelles** [sveltejs/kit#5061]
### Describe the bug

When navigating to a page that has a page endpoint, the following error is shown:

```
500

The string did not match the expected pattern.
json@[native code]
@http://localhost:3000/@fs/Users/philipp/Projects/my-app/.svelte-kit/runtime/client/start.js:1139:56
asyncFunctionResume@[native code]
@[native code]
promiseReactionJobWithoutPromise@[native code]
promiseReactionJob@[native code]
```

When I force a page reload, it works, but navigating to the page never does.

### Reproduction

Create a new skeleton project with kit.trailingSlash = "always". Then create two files.

/src/routes/test.svelte
```svelte
<script>
	/** @type {string} */
	export let title;
</script>

<h1>Titel {title}</h1>
```

/src/routes/test.js
```javascript
/** @type {import('./__types/test').RequestHandler} */
export async function get() {
  return {
    body: {
      title: "test",
    },
  }
}
```

and change /src/routes/index.svelte to
```svelte
<h1>Welcome to SvelteKit</h1>
<p>Visit <a href="https://kit.svelte.dev">kit.svelte.dev</a> to read the documentation</p>
<a href="/test">Test</a>
```

Then navigate to /test using the link to see the error.

### Logs

```shell
Nothing is printed to the browser console or server logs.
```

### System Info

```shell
System:
    OS: macOS 12.4
    CPU: (16) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 2.65 GB / 32.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.15.0 - ~/.nvm/versions/node/v16.15.0/bin/node
    npm: 8.8.0 - ~/node_modules/.bin/npm
  Browsers:
    Chrome: 101.0.4951.64
    Firefox: 100.0.1
    Safari: 15.5
  npmPackages:
    @sveltejs/adapter-static: next => 1.0.0-next.32 
    @sveltejs/kit: next => 1.0.0-next.343 
    svelte: ^3.44.0 => 3.48.0
```

### Severity

blocking an upgrade

**Comment by tcc-sejohnson** [comment]
@Rich-Harris 

Happy to take care of this one if you'd like.

**Comment by mironovich** [comment]
also faced this error.
solved with `headers: { 'Content-Type': 'application/json' }`

**Comment by kristjanmar** [comment]
I also have this problem. I changed it to trailingSlash: 'ignore' for now, but I make sure that all my internal links have a trailing slash and then I have a rel="canonical" tag to make sure search engines know which should be the default.

I believe this bug was introduced in version 328. It worked fine in 327 but not from 328 and after.

**Comment by kristjanmar** [comment]
For those using Vercel, adding "trailingSlash": true to vercel.json seems to fix the problem. Then Vercel enforces the trailing slash.

https://vercel.com/docs/project-configuration#project-configuration/trailing-slash

**Comment by PatrickG** [comment]
I missed this issue before I reported [the same issue](#5148). This should be fixed by #5149.

P.S. Odd error message by safari

**Comment by angelsbv** [comment]
> I missed this issue before I reported [the same issue](https://github.com/sveltejs/kit/issues/5148). This should be fixed by #5149.
> 
> P.S. Odd error message by safari

@Rich-Harris As #5149 is merged this shouldn't be closed?

**Comment by Rich-Harris** [comment]
Yep, looks like this is fixed. Please open a new issue with a reproduction if not
